tests/migration/recovery.go

```diff
@@ -0,0 +1,236 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package migration
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"golang.org/x/net/context"
+
+	batchv1 "k8s.io/api/batch/v1"
+	k8score "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	k8smeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/pkg/util"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/libkubevirt/config"
+	"kubevirt.io/kubevirt/tests/libmigration"
+	"kubevirt.io/kubevirt/tests/libnet/job"
+	"kubevirt.io/kubevirt/tests/libpod"
+	"kubevirt.io/kubevirt/tests/libregistry"
+	"kubevirt.io/kubevirt/tests/libstorage"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/libvmops"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[sig-compute]Migration recovery", decorators.SigCompute, func() {
+	DescribeTable("should successfully defer a migration", func(fakeSuccess, flakeCheck bool) {
+		virtClient, err := kubecli.GetKubevirtClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		if flakeCheck {
+			kv := getCurrentKvConfig(virtClient)
+			var exists bool
+			kv.VMStateStorageClass, exists = libstorage.GetRWOFileSystemStorageClass()
+			Expect(exists).To(BeTrue())
+			config.UpdateKubeVirtConfigValueAndWait(kv)
+		}
+
+		By("Creating a VMI with RWO backend-storage")
+		vmi := libvmifact.NewFedora(
+			libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+			libvmi.WithNetwork(v1.DefaultPodNetwork()),
+		)
+		vmi.Spec.Domain.Devices.TPM = &v1.TPMDevice{
+			Persistent: pointer.P(true),
+		}
+		vmi.Namespace = testsuite.GetTestNamespace(vmi)
+
+		By("Creating a migration policy for that VMI")
+		migrationPolicy := PreparePolicyAndVMIWithNSAndVMILabelsWithPreexistingPolicy(vmi, nil, 1, 0, nil)
+		migrationPolicy.Spec.BandwidthPerMigration = pointer.P(resource.MustParse("1Mi"))
+		CreateMigrationPolicy(virtClient, migrationPolicy)
+
+		By("Starting the VMI as a VM")
+		vm := libvmi.NewVirtualMachine(vmi, libvmi.WithRunStrategy(v1.RunStrategyManual))
+		vm, err = virtClient.VirtualMachine(vmi.Namespace).Create(context.Background(), vm, k8smeta.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		vm = libvmops.StartVirtualMachine(vm)
+		vmi = libwait.WaitForVMIPhase(vmi, []v1.VirtualMachineInstancePhase{v1.Running})
+		Expect(console.LoginToFedora(vmi)).To(Succeed())
+		pod, err := libpod.GetPodByVirtualMachineInstance(vmi, vmi.Namespace)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Stressing")
+		runStressTest(vmi, stressDefaultVMSize, 42)
+
+		By("Starting a slow migration")
```

Comment:
For CPU hotplug with start a slow migration which never converges using the migration policy and very low bandwidth. See [here](https://github.com/kubevirt/kubevirt/blob/main/tests/hotplug/cpu.go#L327). Maybe this consumes less resource on the CI

Reply:
That looks the same as what I'm doing, or am I missing something?

Reply:
mmh, we don't run any stress test inside the vm there

Reply:
Ah sorry, I missed the line above, sorry!!